extension/extension.go

```diff
@@ -43,6 +43,10 @@ type PipelineWatcher interface {
 	NotReady() error
 }
 
+type ConfigWatcher interface {
+	ConfigResolved(data map[string]interface{}) error
```

Comment:
Why not passing context? Why not passing the confmap.Conf?

Reply:
> Why not passing context?

I didn't pass context since I was mostly following the approach used by the `PipelineWatcher` interface, but I have no issue doing it here. I will add it.

> Why not passing the confmap.Conf?

I went for just passing a map since I don't think extensions will need any functionality from confmap.Conf. I see the value in using a Collector-specific configuration structure though, so I can change to that for now.